WordPress/WordPressTest/TaggedManagedObjectIDTests.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import XCTest
+
+@testable import WordPress
+
+class TaggedManagedObjectIDTests: CoreDataTestCase {
+
+    func testQuerySaved() throws {
+        let post = PostBuilder(contextManager.mainContext).with(title: "Test post").build()
+        try contextManager.mainContext.save()
+
+        let id = TaggedManagedObjectID(saved: post)
+        let result = try contextManager.mainContext.existingObject(with: id)
+        XCTAssertEqual(result.postTitle, "Test post")
+    }
+
+    func testQueryUnsaved() throws {
+        let post = PostBuilder(contextManager.mainContext).with(title: "Test post").build()
+        let id = try TaggedManagedObjectID(unsaved: post)
+
+        try contextManager.mainContext.save()
+
+        let newContext = contextManager.newDerivedContext()
+        let result = try newContext.existingObject(with: id)
+        XCTAssertEqual(result.postTitle, "Test post")
+    }
+
+    func testQueryUnsavedUsingTheSameContext() throws {
+        let context = contextManager.mainContext
+        let post = PostBuilder(context).with(title: "Test post").build()
+        let id = try TaggedManagedObjectID(unsaved: post)
+
+        let result = try context.existingObject(with: id)
+        XCTAssertEqual(result.postTitle, "Test post")
+    }
+
+    func testQueryUnsavedUsingDifferentContext() throws {
+        let post = PostBuilder(contextManager.mainContext).build()
+        let id = try TaggedManagedObjectID(unsaved: post)
+
+        let newContext = contextManager.newDerivedContext()
+
+        let expectation = expectation(description: "The query should fail because the model is not saved")
+        do {
+            _ = try newContext.existingObject(with: id)
+        } catch {
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.1)
```

Comment:
What do you think of

```suggestion
        XCTAssertThrowsError(try newContext.existingObject(with: id), "The query should fail because the model is not saved")
```

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/0d1a3267-7a2d-4abb-b481-c21104eadaaa)

Reply:
Neat! I'll take your code suggestion.